src/Common/tests/Tests/System/Net/VirtualNetworkStreamTest.cs

```diff
@@ -109,10 +119,19 @@ public void VirtualNetworkStream_MultiThreadIntegrityTest_Ok()
             {
                 Parallel.For(0, 100000, async (int i) =>
```

Comment:
Thanks for pointing that out! This is not obvious at all.
Looks like this test was not doing anything before. I'm now seeing the perf impact caused by Task.Run() used in the ReadAsync/WriteAsync.